src/session.rs

```diff
@@ -130,10 +137,11 @@ pub fn start(config: &Config, initial_request: Option<String>) -> i32 {
                         // get didClose message without running controller, unless it crashed
                         // before. In that case didClose can be safely ignored as well.
                         if request.method != notification::DidCloseTextDocument::METHOD {
-                            debug!("Spawning a new controller for {:?}", route);
+                            debug!("Spawning a new controller for {:#?}", routes);
                             controller_entry.insert(spawn_controller(
                                 config.clone(),
-                                route,
+                                language_id.clone(),
```

Comment:
Haven't had time for a full review, some comments in passing:
I'm surprised we only spawn a controller for the first server.
Won't that lead to weird behavior if the configured servers have different root paths or language IDs?

Reply:
It's not exactly for the first server, as the `controllers` key is now a vector of routes, and not a single route anymore. So in reality it spawns a controller for the first time a route sequence is seen by kak-lsp.